app/services/idv/data_url_image.rb

```diff
@@ -1,16 +1,32 @@
 module Idv
   class DataUrlImage
     def initialize(data_url)
-      @header, base64_data = data_url.split(',', 2)
-      @data = Base64.decode64(base64_data || '')
+      header, data = URI(data_url.chomp).opaque.to_s.split(',', 2)
+      @header = header.to_s
+      @data = data.to_s
     end
 
+    BASE64_CONTENT_TYPE = /;base64$/.freeze
+
+    # @return [String]
     def content_type
-      @header.gsub(/^data:/, '').gsub(/;base64$/, '')
+      content_type, *_rest = @header.split(';')
+      content_type.to_s
     end
 
+    # @return [String]
     def read
-      @data
+      if base64_encoded?
+        Base64.decode64(@data)
+      else
+        CGI.unescape(@data)
```

Comment:
Curious of the behavior here, I [looked it up](https://devdocs.io/ruby~2.6/cgi/util#method-i-unescape) and noticed this also handles conversion of `+` to space `' '`, which I think is not what we should expect.

```js
decodeURIComponent('Hello%2C%20World!');
// "Hello, World!"

decodeURIComponent("%27Stop%21%27+said+Fred");
// "'Stop!'+said+Fred"
```

But it is what we'd get from `URI::decode`:

```rb
URI::decode('Hello%2C%20World!')
# "Hello, World!"

URI::decode("%27Stop%21%27+said+Fred")
# "'Stop!'+said+Fred"
```

```suggestion
        URI.decode(@data)
```

Reply:
Ah good catch, thanks!

Reply:
Rubocop doesn't like `URI.decode` but it point me towards `decode_www_form_component` which sounds much closer to what we want

Reply:
Moved this over to #4074